src/tir/transforms/update_pointer_storage_scope.cc

```diff
@@ -0,0 +1,91 @@
+/*
```

Comment:
I think we need to also update `lower_device_storage_access_info.cc` to use the PointerType storage scope in this PR as well.

With a custom storage scope used in a cache_read stage, e.g. 
```
s.cache_read(data, "global.texture", [OL])
```

I see that [lower_device_storage_access_info.cc#L72](https://github.com/apache/tvm/blob/main/src/tir/transforms/lower_device_storage_access_info.cc#L72) is unhappy with the scope tag.  
```
  Check failed: (e.info.defined()) is false: Cannot find memory info of global.texture
```

Reply:
Hmm interesting. `attr::storage_scope` and the PointerType should have the same storage scope, by the invariant at https://github.com/apache/tvm/blob/3e5e0efe2dfde96f9b3a162ff66524ff79b62723/src/tir/ir/stmt.cc#L64-L73. 

A possible explanation is that the value of `scope.tag.length()` at https://github.com/apache/tvm/blob/main/src/tir/transforms/lower_device_storage_access_info.cc#L70 is different between `main` and this branch. In this case, `GetMemoryInfo` assumes that a packed function `tvm.info.mem.global.texture` is defined, but I don't find it anywhere.

Do you have a repro script?

Reply:
Thanks @masahi. I tracked this issue down to the buffer storage scope being set to `global` and removing the scope tag. Previously this was fine as the AttrStmtNode realize_scope annotation was used during texture lowering. 

This demonstrate the value in removing the buffer storage scope in favor of consolidating around the PointerType's storage_scope. I look forward to that PR! Thanks again for this change.